src/app/services/reducers/resources-reducer.ts

```diff
@@ -20,20 +24,38 @@ export function resources(state: IResources = initialState, action: IAction): IR
       return {
         pending: false,
         data: action.response,
-        error: null
+        error: null,
+        paths: []
       };
     case FETCH_RESOURCES_ERROR:
       return {
         pending: false,
         error: action.response,
-        data: res
+        data: res,
+        paths: []
       };
     case FETCH_RESOURCES_PENDING:
       return {
         pending: true,
         data: initialState.data,
-        error: null
+        error: null,
+        paths: []
       };
+    case RESOURCEPATHS_ADD_SUCCESS:
+      const paths: IResourceLink[] = state.paths || [];
+      action.response.forEach((element: any) => {
+        const exists = !!paths.find(k => k.key === element.key);
+        if (!exists) {
+          paths.push(element);
```

Comment:
Performing side effects in a reducer leads to hard to debug bugs. You can avoid the side-effect with `paths = [...paths, element]` which creates a new copy of the variable.